### DISCUSSION TITLE
nidigital：write_source_waveform_broadcast(waveform_name, waveform_data) waveform_data type is?

### DISCUSSION THREAD

**Discussion by sbethur** [ni/nimi-python#1616] [Q&A]
@super-leilei asked in #1613:

> session.pins['RFFEDATA'].create_source_waveform_serial('RegWrite',data_mapping,1,bit)
session.write_source_waveform_broadcast('RegWrite',[1,0,0,0,1,1,1,0,0,1,0,0,1,0,0,1,1,0,1])
Whether the above method is correct, trouble to inform, thank you!

**Comment by sbethur** [comment]
Method parameter types can be found in nimi-python [documentation](https://nimi-python.readthedocs.io/en/master/).

As per the [documentation](https://nimi-python.readthedocs.io/en/master/nidigital/class.html#write-source-waveform-broadcast), `waveform_data` type is `list of int`.

>  session.write_source_waveform_broadcast('RegWrite',[1,0,0,0,1,1,1,0,0,1,0,0,1,0,0,1,1,0,1])
Whether the above method is correct, trouble to inform, thank you!

This is correct.

**Reply by Super-leilei** [reply]
Does the value of "waveform_data" correspond to the part circled in the "NI Digital Pattern Editor" as shown in the figure above? Thanks!
![image](https://user-images.githubusercontent.com/84898853/119915143-6ece2200-bf94-11eb-9f73-4d2ff1bc7b53.png)

**Reply by Super-leilei** [reply]
@sbethur

**Reply by sbethur** [reply]
Yes, that is correct.
